src/Package/Impl/Debugger/DataTips/DataTipTextViewFilter.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using Microsoft.Languages.Editor.Composition;
+using Microsoft.Languages.Editor.Text;
+using Microsoft.R.Components.ContentTypes;
+using Microsoft.R.Core.DataTips;
+using Microsoft.R.Editor.Document;
+using Microsoft.VisualStudio.Editor;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.TextManager.Interop;
+
+namespace Microsoft.VisualStudio.R.Package.Debugger.DataTips {
+    internal class DataTipTextViewFilter : IOleCommandTarget, IVsTextViewFilter {
+        private readonly IVsEditorAdaptersFactoryService _adapterService;
+        private readonly IVsDebugger _debugger;
+        private readonly IWpfTextView _textView;
+        private readonly IVsTextLines _vsTextLines;
+        private readonly IOleCommandTarget _nextTarget;
+
+        public DataTipTextViewFilter(IWpfTextView textView, IVsDebugger debugger) {
+            if (!textView.TextBuffer.ContentType.IsOfType(RContentTypeDefinition.ContentType)) {
+                return;
+            }
+
+            _textView = textView;
+            _debugger = debugger;
+            _adapterService = ComponentLocator<IVsEditorAdaptersFactoryService>.Import();
```

Comment:
I'll move it to the caller. It's inconsistent now in that the caller retrieves debugger service, but this thing retrieves the adapter service.